lib/shared/puppet_forge/module_name.rb

```diff
@@ -0,0 +1,23 @@
+module PuppetForge
+  # Normalize Forge module names
+  #
+  # Forge modules use '-' as the separator between the username and modulename
+  # in the module slug, but the Puppet Forge uses '/' as the separator on module
+  # pages and in the URL. This class hides the distinction by always using '-' as
+  # the separator for normal use. If the name needs to use the slash separated name
+  # then the #forge_name method can be used.
+  #
+  # @api private
+  class ModuleName < String
+
+    def initialize(*args)
+      super
+      tr!('/', '-')
+    end
+
+    # @return [String] The slash separated module name.
+    def forge_name
+      String.new(self).tr('-', '/')
+    end
```

Comment:
That's a good point actually - the oddness that I pointed out above that required the explicit cast to string was a demonstration of this quirk. I'll give the Forwardable approach a once over to see if it's something we can stuff in really quickly.